### PR TITLE
Add Adventure components title and originalTitle in InventoryView

### PR DESCRIPTION
**Pull request by LeonidMem**
Honestly, I am not entirely sure if I changed the method CraftInventoryView.sendInventoryTitleChange(InventoryView, String) correctly for future work, because I decided to move from duplicated code, so, please, let me know how it is better to resolve such things.

### CODE REVIEW

File: patches/server/0994-Add-Adventure-components-title-and-originalTitle-in-.patch [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
++            }
++
++            @Override
++            public net.kyori.adventure.text.Component originalTitle() {
++                return this.originalTitleComponent;
+             }
+             // Paper end
+ 
+             @Override
++            @Deprecated // Paper
```
Comment (lynxplay) [inline patches/server/0994-Add-Adventure-components-title-and-originalTitle-in-.patch:44]: No need to annotate internal methods for deprecation.
(Repeats in a few spaces in the patch)

File: patches/server/0994-Add-Adventure-components-title-and-originalTitle-in-.patch [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+             public String getOriginalTitle() {
+                 return originalTitle;
+             }
+ 
+             @Override
++            @Deprecated // Paper
+             public void setTitle(String title) {
+                 CraftInventoryView.sendInventoryTitleChange(this, title);
```
Comment (lynxplay) [inline patches/server/0994-Add-Adventure-components-title-and-originalTitle-in-.patch:58]: Wouldn't it be smart to just invoke title() here with the result of the legacy deserialisation ?

File: patches/server/0994-Add-Adventure-components-title-and-originalTitle-in-.patch [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+     public String getOriginalTitle() {
+         return this.originalTitle;
+     }
+ 
+     @Override
++    @Deprecated // Paper
+     public void setTitle(String title) {
+         CraftInventoryView.sendInventoryTitleChange(this, title);
+         this.title = title;
++        this.titleComponent = net.kyori.adventure.text.serializer.legacy.LegacyComponentSerializer.legacySection().deserialize(this.title); // Paper
```
Comment (lynxplay) [inline patches/server/0994-Add-Adventure-components-title-and-originalTitle-in-.patch:127]: Same here, probably best to just delegate this to the paper method to reduce duplicates.

File: patches/server/0994-Add-Adventure-components-title-and-originalTitle-in-.patch [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+     }
+ 
+     public boolean isInTop(int rawSlot) {
+@@ -104,7 +127,16 @@ public class CraftInventoryView extends InventoryView {
+         return this.container;
+     }
+ 
++    @Deprecated // Paper
+     public static void sendInventoryTitleChange(InventoryView view, String title) {
++        // Paper start
```
Comment (lynxplay) [inline patches/server/0994-Add-Adventure-components-title-and-originalTitle-in-.patch:137]: This entire block can be made a bit pretty to look more like this:
```patch
     public static void sendInventoryTitleChange(InventoryView view, String title) {
+    // Paper start
+        Preconditions.checkArgument(title != null, "Title cannot be null");
+        CraftInventoryView.sendInventoryTitleChange(view, net.kyori.adventure.text.serializer.legacy.LegacyComponentSerializer.legacySection().deserialize(title));
+    }
+
+    public static void sendInventoryTitleChange(InventoryView view, net.kyori.adventure.text.Component title) {
+    // Paper end
         Preconditions.checkArgument(view != null, "InventoryView cannot be null");
         Preconditions.checkArgument(title != null, "Title cannot be null");
         Preconditions.checkArgument(view.getPlayer() instanceof Player, "NPCs are not currently supported for this function");
@@ -113,7 +145,8 @@ public class CraftInventoryView extends InventoryView {
         final ServerPlayer entityPlayer = (ServerPlayer) ((CraftHumanEntity) view.getPlayer()).getHandle();
         final int containerId = entityPlayer.containerMenu.containerId;
         final MenuType<?> windowType = CraftContainer.getNotchInventoryType(view.getTopInventory());
-        entityPlayer.connection.send(new ClientboundOpenScreenPacket(containerId, windowType, CraftChatMessage.fromString(title)[0]));
+        entityPlayer.connection.send(new ClientboundOpenScreenPacket(containerId, windowType, io.papermc.paper.adventure.PaperAdventure.asVanilla(title))); // Paper 
         ((Player) view.getPlayer()).updateInventory();
     }

**Review by lynxplay — CHANGES_REQUESTED** [review]
Welcome to paper :tada: grats on your first PR!

A few comments in the patch, beyond that generally, this change should live in the existing Adventure patch instead of being its own patch.
See the CONTRIBUTING.md on how to modify an existing patch.

**Comment by Machine-Maker** [comment]
This is actually a duplicate PR (https://github.com/PaperMC/Paper/pull/9330). Since that one was made first, and they do the exact same thing, I'm going to close this one.

**Pull request closed without merging**